server/apps/immich/src/api-v1/asset/asset.service.ts

```diff
@@ -189,62 +196,21 @@ export class AssetService {
     return this.downloadService.downloadArchive(`immich-${now}`, assetToDownload);
   }
 
-  public async downloadFile(query: ServeFileDto, assetId: string, res: Res) {
-    try {
-      let fileReadStream = null;
-      const asset = await this._assetRepository.getById(assetId);
-
-      // Download Video
-      if (asset.type === AssetType.VIDEO) {
-        const { size } = await fileInfo(asset.originalPath);
-
-        res.set({
```

Comment:
I don't recal why I had to set the content-type and content-length here. But the testing went well even without these two header get explicitly set. Did you encounter anything not working right without the two headers get set?

Reply:
So these actually do get set, just through the options in the streamable file object. I'm trying to keep transport specific stuff like this outside of the "domain". The domain returns the value, then the controller maps the response into a streamable file. The second argument ends up being set as a header.

Reply:
![image](https://user-images.githubusercontent.com/27055614/216636328-97b125b1-20d1-4eba-97e7-dce7aa2023e8.png)
Through this?

Reply:
Yes